micro/reify_test.go

```diff
@@ -28,44 +27,44 @@ scheme code:
 	)
 */
 func TestReify(t *testing.T) {
-	a1 := "(,x . (,u ,w ,y ,z ((ice) ,z)))"
-	a2 := "(,y . corn)"
-	a3 := "(,w . (,v ,u))"
-	s := "(" + a1 + " " + a2 + " " + a3 + ")"
-	e, err := sexpr.Parse(s)
-	if err != nil {
-		t.Fatal(err)
-	}
+	u := ast.NewVar("u", 0)
```

Comment:
`u := mirco.NewVar()` could probably also be fine right?
Then it generates a number and string and the user simply has to worry about passing the Go variable `u` to the right spots in the microkanren expression.  What do you think?

Reply:
Would've been nice but we need to guarantee order or get flaky tests.
Reify sorts on var index now :)

Reply:
Okay lets think about this again, at a later commit